src/p2p_async.rs

```diff
@@ -83,8 +83,8 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> P2psConnAsync<T> {
         Ok(())
     }
 
-    /// Reads data from a stream decrypts it returning the data
-    pub async fn read(&mut self) -> std::io::Result<Vec<u8>> {
+    /// Reads data from a stream decrypts it returning the data and len
+    pub async fn read_len(&mut self) -> std::io::Result<(Vec<u8>, usize)> {
```

Comment:
no need for this function to be public, it could create confusion
```suggestion
    async fn read_len(&mut self) -> std::io::Result<(Vec<u8>, usize)> {
```

Reply:
only used internally so yeah.